### PR TITLE
Forgot password feature

### PR DESCRIPTION
**Pull request by federico-barbieri**
Good news, this works.

If you want to test it out, remember to add your own environmental variables in the backend. This is needed to send emails using nodemailer. You will need your email and a special password to be used with third party applications. I think we used it with Santiago, where google gives us a 16 character password.

As a very simple explanation:

The login form now has a "forgot password?" link that takes you to a new page "ForgotPassword". Here you can put your email and if you are in the database then you will receive an email that has a token attached to it. 

I created a new Model in the database that links tokens with users. So when you click the link, you are redirected to a new page, "resetPassword". you add your password and repeat it. 

We grab the token and if it exists in the db, we grab the user associated with it and that should do the trick.

I think it's a cool feature to have and also cool to talk about the process behind it.

### CODE REVIEW

**Review by henrynavntoft — CHANGES_REQUESTED** [review]
Looks super nice! Just a few places there are some Type issues, mostly in the authAPI in the frontend, which results in the frontend crashing in prod. 

Here are the logs from vercel: 
src/api/authApi.tsx(2,38): error TS2305: Module '"../types"' has no exported member 'resetPassword'.
src/api/authApi.tsx(2,38): error TS2395: Individual declarations in merged declaration 'resetPassword' must be all exported or all local.
src/api/authApi.tsx(18,14): error TS2395: Individual declarations in merged declaration 'resetPassword' must be all exported or all local.
src/api/authApi.tsx(18,75): error TS2304: Cannot find name 'ResetPasswordRequest'.
Error: Command "npm run build" exited with 2

File: frontend/src/api/authApi.tsx [hunk 1]
```diff
@@ -9,6 +9,17 @@ export const loginUser = async (loginData: LoginPayload): Promise<AuthResponse>
...
 };
 
+// forgot password
+export const forgotPassword = async (email: string): Promise<void> => {
+  await axiosInstance.post('/api/auth/forgotpassword', { email }); 
+};
+
+// reset password
+export const resetPassword = async ({ token, password, confirmPassword }: ResetPasswordRequest): Promise<void> => {
+  await axiosInstance.post('/api/auth/resetpassword', { token, password, confirmPassword });
```
[the remark this answers is not in the document]
Comment (bionordic) [reply frontend/src/api/authApi.tsx:19]: I think this error is just a duplicate of the one above. 

The endpoint is /api/auth/resetpassword, not resetpasswordpayload

// Reset Password - Validate Token and Update Password
router.post('/resetpassword', async (req, res) => {}

[the remark this answers is not in the document]
Comment (federico-barbieri) [reply frontend/src/api/authApi.tsx:19]: wow didn't know you could access the repo from another account associated to the one in the repo. crazy world

File: frontend/src/api/authApi.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import axiosInstance from './axiosInstance';
-import { AuthResponse, LoginPayload } from '../types';
+import { AuthResponse, LoginPayload, resetPassword } from '../types';
```
[the remark this answers is not in the document]
Comment (federico-barbieri) [reply frontend/src/api/authApi.tsx]: good catch

**Pull request closed without merging**